casper/src/main/scala/coop/rchain/casper/Estimator.scala

```diff
@@ -166,9 +182,10 @@ object Estimator {
       blockDag: BlockDagRepresentation[F],
       scores: Map[BlockHash, Long]
   ): F[List[BlockHash]] =
-    for {
-      c <- blockDag.children(b).map(_.getOrElse(Set.empty[BlockHash]).filter(scores.contains))
-    } yield if (c.nonEmpty) c.toList else List(b)
+    blockDag.children(b).map(_.getOrElse(Set.empty[BlockHash]).filter(scores.contains)).map {
```

Comment:
First of all this line is unreadable. One idea to improve it quite easily is to extract a named function with the names of the insides of the maps.

Second, this could be a private function in sortChildren so you don't have to pass explicitely  F, blockDag and the scores.

Third, I find the original `if ... else ...` less verbose and more readable than the new `case ... if ... case...`.

Reply:
I like seeing what is explicitly passed in. I'll make the function more readable though.